libs/entitlement/bridge/src/main/java/org/elasticsearch/entitlement/bridge/EntitlementChecker.java

```diff
@@ -254,4 +261,73 @@ public interface EntitlementChecker {
     void check$java_net_Socket$connect(Class<?> callerClass, Socket that, SocketAddress endpoint);
 
     void check$java_net_Socket$connect(Class<?> callerClass, Socket that, SocketAddress endpoint, int backlog);
+
+    /* NIO
+     * For NIO, we are sometime able to check a method on the public surface/interface (e.g. AsynchronousServerSocketChannel#bind)
+     * but most of the time these methods are abstract in the public classes/interfaces (e.g. ServerSocketChannel#accept,
+     * NetworkChannel#bind), so we are forced to implement the "impl" classes.
+     * You can distinguish the 2 cases form the namespaces: java_nio_channels for the public ones, sun_nio_ch for the implementation
+     * classes. When you see a check on a sun_nio_ch class/method, this means the matching method on the public class is abstract
+     * (not instrumentable).
+     */
+
+    // bind
+
+    void check$java_nio_channels_AsynchronousServerSocketChannel$bind(
```

Comment:
Wondering, should we consistently instrument the impl for simplicity? `bind` on the abstract class is final and redirects to the impl... on the other hand it feels like relying on implementation details.

Reply:
IDK, it's true that we are already relying on implementation details for anything else, but ideally we would stuck with the public interface whenever this is possible.